### DISCUSSION TITLE
Title: Idea - Extreme Long Timelapse

### DISCUSSION THREAD

**Discussion by TikDesign** [gopro/labs#658] [General]
I have an idea to create a programmable DC/DC power USB supply for GoPro.

Functions: Input voltage in the range of DC 6V-36V
Output voltage DC: 5V
Maximum output current: 2.5A
DC/DC Automotive Step-Down Voltage Regulator: e.g. LM636x5-Q1
Battery capacity: unlimited

The timer should have its own small battery and use minimal power. The timer will be made from a power-saving microcontroller from Silicon Labs. The program should be created to be interrupt-based, utilizing one of the MCU's timers to control the RTC. Maximum power consumption: 150uA.
The power supply, when disabled, should only draw a few uA from the battery.
The first version will be made such that power is supplied to the USB once every 24 hours, for example between 09:00 - 15:00. A possible extension would be that it can set outputs according to a schedule. For example, 09:00 - 10:30 & 12:00-15:00 & 20:00-22:00.

Does anyone have thoughts on this? Maybe there is a product like that somewhere?

**Comment by dnewman-gpsw** [comment]
Likely unnecessary at least for HERO12 and future cameras.  The low power MCU on H12 is pretty efficient and the reliability so greatly improved, I'm been testing multiple cameras running for weeks, with only camera, Voltaic V50 battery, and solar charger for the external battery. The are some Labs improvements to help on H12 coming out in a few weeks. I plan to document more then.

**Comment by TikDesign** [comment]
I want to take pictures every other second over a 6-hour interval, each day for a total of 14 days. Does the Hero 12 support such functionality? Is the maximum duration still 3 hours?

**Reply by dnewman-gpsw** [reply]
Sure, pretty easy with Labs.  There is duration limit if you don't set a limit.

!09:00NmTPp.2!S!21600E!1R 

This will start at 9am, record timelapse photos for 6 hours, then repeat the next day, forever. Use USB power.

**Reply by TikDesign** [reply]
I think I misunderstood something. I had set the duration inside the camera to 3 hours instead of no limits. In the QR code, it was 6 hours. So I assume that's why the camera stopped taking pictures after 3 hours and not 6 hours. Thank you very much for your information.

**Reply by dnewman-gpsw** [reply]
!09:00NmTPp.2dR0!S!21600E!1R  Added dR0 to set duration record to unlimited (it is the default setting.)

**Comment by Espusen** [comment]
This is really interresting!
I am trying to do the same thing with a pico and a RTC. I will need to test for reliability.

The goal is to have RELIABLE long term timelapse. All test I have done with LABS alone fails after 2-4 days.
I am hoping that the BOOT script will be infinately more reliable. Does anyone have any experience with this?

Having the USB power to be sequenced on, picture taken, then shut down.

Thank you for all help!

Espus

**Reply by dnewman-gpsw** [reply]
This is very camera model/FW dependent. On HERO12 I've run in for weeks with no issue. I expect HERO12 without an external controller will be more reliable.

**Reply by Espusen** [reply]
Wow, you are a better man than me then.
I have been testing Hero 9 and 10 for a loooong while to try to get this to work. Admittedly I stopped testing in June 23 and I have not tested the -11 or -12.
So you have tested Labs ONLY on a 12 and had it running for several weeks?

But with it being camera dependant it is not good enough for professional work. I need it to be reliable. That is why I want to try the timer option, as it is not reliant on the GoPro itself keeping time. (Which alledgedly is the problem..?)

**Reply by dnewman-gpsw** [reply]
Yes, for an external trigger, than can be made camera independent.  The HERO12 solution is just the simplest, easy to weather proof (even if you provide external battery and solar power.  All configured through a QR Code.

**Reply by Espusen** [reply]
Has this been tested for long term reliability??
For the HERO 9 TL code the best I achieved was 8 days running.

**Reply by dnewman-gpsw** [reply]
Shorter interval cause failures faster. So in testing I might use 60s or a 3-4 minutes interval. I run three HERO12s this way for about week with no failures. I'd never seen anything like this reliability. Typically something random would happen between 400-500 captures. On HERO12 no failures at thousands scripted captures on multiple cameras.  The most recent tests included several uploads a day, auto upload that was what I was testing for the next Labs release (pictured cloud result from three cameras.)

For reliability in these tests, these all started with a 100% full battery internally, and Voltaic V50 external battery.  Have an internal full battery to start solves a potential reliability issue, as it avoids the camera transition from fast charging to trickle charging.

![image](https://github.com/gopro/labs/assets/10856084/cdad1b28-8236-4b93-a8c7-8591cb0b2414)

**Reply by Espusen** [reply]
Wow that's amazing!!
Very positive news! I will be getting a H12 and start testing this immediately after christmas.

**Comment by Espusen** [comment]
I actually make and sell weatherproofing kits for this purpose. It just didnt work with the HERO9 or 10 unless you used the UpBLink controller..
![Promo profile1 3000](https://github.com/gopro/labs/assets/153327756/6fd6df4c-0529-4c57-b57e-e2bf263cd86b)

**Reply by dnewman-gpsw** [reply]
Nice. You need to update the website to list HERO12.

**Reply by Espusen** [reply]
How do you mean??

**Reply by dnewman-gpsw** [reply]
Is this your site: 
![image](https://github.com/gopro/labs/assets/10856084/282166a9-5124-4bdb-851d-b719c504193e)

HERO12 is not listed

**Reply by Espusen** [reply]
Thanks for noting this. I will amend it today

**Reply by dnewman-gpsw** [reply]
Same issue on the order page.

**Comment by Tocio88** [comment]
Hi everybody, here at University in Utrecht i'm setting up a GoPro12 to take picture every 2h forever for a project in Greenland to have a timelapse of the glaciers over there. 
I have installed the GoPro Lab firmware with this QR code: !7200NmPNr0fWeAoSMwAsMaLcNb1d1i64M1S0gOdPoV10B10W0oDOv0q0!S!R!1R as well the !WAKE=2 and !TUSB=1 because i have a voltaic v75 with solar panel.
It works but i'm seeing that the timestamp into the metadata of the pictures isn't right. All the pictures get the same timestamp, and that's the timestamp when I started scanning the QRcode. Also when the camera wakes up it shows the time, and the remaining time till the next wake up, and keeps the same all the time. Seems doesn't update the internal RTC.
Am I doing something wrong with the code, or some of you experienced the same?

**Reply by dnewman-gpsw** [reply]
This is not correct behavior, however I can list some potential causes.  The log file would help from MISC/qrlot.txt

1) You haven't mentioned whether you have the internal battery installed?  Internal battery helps keep the RTC alive.
2) That command is crazy, you are setting everything, every time the camera boots, you shouldn't do that. Also you have two repeat commands.  Set your settings once. Do this as a separate command: **mPNr0fWeAoSMwAsMaLcNb1d1i64M1S0gOdPoV10B10W0oDOv0q0**

Then your script is just: **!7200NmPN!S!5R**  (I added a few more seconds for Night Photo to complete before shutdown.)

3) Most likely: If the time-date is the same as your start, I expect you had "Set date + time automatically", which means the camera is setting the same time every loop.  User error.

**Reply by Tocio88** [reply]
Here I have the log file: [qrlog.txt](https://github.com/gopro/labs/files/13658271/qrlog.txt)

1. yes, the battery is installed because indeed I read that is better for the RTC
2. I have set the long command, and then the short one now as you suggested. I will leave it run today and check the picture tomorrow.
3. probably, but where is this button to click to make it? Is the "set date and time" button that there is on the side on the QR code of the app? and because I had it in the long command was doing it at every startup?

**Reply by dnewman-gpsw** [reply]
The qrlog.txt confirmed you are setting date and time in the QR code.

[2023-12-11 10:35:35] Read QR:**oT231212123709.495oTD0oTZ1oTI50485**!7200NmPNr0eAoSMwAsMaLcNb1d1i64M1S0g0dPoV1oB1oW0oD0v0q0!S!1R 

Don't use "set date and time" button

**Reply by Tocio88** [reply]
yess! Thanks a lot!
Got it running, with good timestamp but now I did the test to empty both internal and external battery and at startup, when connecting the voltaic v75 it shows me a banner saying that the camera has been shut down due to empty battery and it needs to press ok to remove it and go on. 
Is there a way to avoid this?

**Reply by dnewman-gpsw** [reply]
Maybe.  !MBYPS=1 bypasses many system messages, I just have confirmed if this is one of them.  Obviously it is best to not let the batteries completely run out.
